src/index.js

```diff
@@ -0,0 +1,159 @@
+    const hsl = this._HSVtoHSL(h, s, b);
+
+    return {
+      hsl: {
+        raw: hsl,
+        formatted: `hsl(${hsl[0]}, ${hsl[1]}%, ${hsl[2]}%)`
+      }
+    };
+  }
+
+  _pickHue = () => {
+    let hue = this._pseudoRandom([0, this.HUE_MAX]);
+    const min = hue % this.HUE_MAX;
+    const max = (hue + 1) % this.HUE_MAX;
+
+    hue = this._pseudoRandom([min, max]);
+
+    // Red is on both ends of the color spectrum. Instead of storing red as two
+    // ranges, lookup is grouped in colorDictionary using negative numbers.
+    if (hue < 0) hue = this.HUE_MAX + hue;
+
+    return hue;
+  }
+
+  _pickSaturation = (h) => {
+    const saturationRange = this._getColorInfo(h)[2];
+    let min = saturationRange[0];
+    let max = saturationRange[1];
+    return this._pseudoRandom([min, max]);
+  }
+  
+
+  _pickBrightness = (h, s) => {
+    let min = this._getMinimumBrightness(h, s);
+    let max = 100;
+    return this._pseudoRandom([min, max]);
+  }
+
+  _getMinimumBrightness = (h, s) => {
+    const lowerBounds = this._getColorInfo(h)[1];
+    for (let i = 0; i < lowerBounds.length - 1; i++) {
+      const s1 = lowerBounds[i][0];
+      const v1 = lowerBounds[i][1];
+      const s2 = lowerBounds[i + 1][0];
+      const v2 = lowerBounds[i + 1][1];
+      if (s >= s1 && s <= s2) {
+        const m = (v2 - v1) / (s2 - s1);
+        const b = v1 - m * s1;
+        return m * s + b;
+      }
+    }
+    return 0;
+  }
+
+  // A linear congruential generator (LCG) algorithm that yields a sequence of
+  // pseudo-randomized numbers calculated with a discontinuous piecewise linear
+  // equation. see: indiegamr.com/generate-repeatable-random-numbers-in-js
+  _pseudoRandom = (range) => {
+    const max = range[1] || 1;
+    const min = range[0] || 0;
+    this.seed = (this.seed * 9301 + 49297) % 233280;
+    const rnd = this.seed / 233280;
+    return Math.trunc(min + rnd * (max - min));
+  }
+
+  _getColorInfo = (hue) => {
+    // Red is on both ends of the color spectrum. Map them together:
+    if (hue >= 334 && hue <= this.HUE_MAX) {
+      hue -= this.HUE_MAX;
+    }
+
+    return this._colorDictionary.find((c) => hue >= c[0][0] && hue <= c[0][1]);
+  }
+
+  _HSVtoHSL = (h, s, v) => {
+    const round = (num) => Math.trunc((num + Number.EPSILON) * 100) / 100;
+    const l = (2 - s / 100) * v / 2;
+    let saturation = s * v / (l < 50 ? l * 2 : 200 - l * 2);
+
+    // Handle division-by-zero
+    if (isNaN(saturation)) saturation = 0;
+
+    return [
+      h,
+      round(saturation),
+      round(l)
+    ];
+  }
+
+  _stringToInteger = (string) => {
+    let total = 0;
+    for (let i = 0; i !== string.length; i++) {
+      if (total >= Number.MAX_SAFE_INTEGER) break;
+      total += string.charCodeAt(i);
+    }
+    return total;
+  }
+
+  // From https://github.com/Evercoder/d3-color-difference
+  _euclideanDistance = (a, b) => {
+    return Math.sqrt(Math.pow(a[0] - b[0], 2) + Math.pow(a[1] - b[1], 2) + Math.pow(a[2] - b[2], 2));
```

Comment:
This seems to work ok but is it odd that we are equally weighting hue, saturation, and lightness. HSL isn't a color space designed to support measuring euclidean distances in this way. Can we instead use RGB or LAB? According to https://stackoverflow.com/questions/3231410/distance-measure-between-hsl-colours, LAB is designed for more easily measuring color distances.